python/ray/serve/_private/replica.py

```diff
@@ -636,6 +637,12 @@ async def reconfigure(
                     deployment_config.user_config
                 )
 
+            # We need to update internal replica context to reflect the new
+            # deployment_config and servable_object.
+            self._set_internal_replica_context(
+                servable_object=self._user_callable_wrapper.user_callable
+            )
+
```

Comment:
hmm why is this necessary?

Reply:
oh I see, this is also updating the deployment_config. The comment about `servable_object` is irrelevant here

Reply:
Sg, will drop the comment about `servable_object`👍